noobaa/noobaa-core
# Allow customizing keep-alive and request timeouts for endpoint servers

### Explain the changes
This PR adds 2 configurations to endpoints, the intention is to allow the users to configure these timeouts based on their environments. For example, in a case where a NooBaa deployment is behind a horribly excruciatingly slow VPN, in those cases the user might want to tweak these values. 

**NOTE:** This still requires cooperation from the AWS client. So it is important for the client to use `--cli-read-timeout` and `--cli-connect-timeout`.

### Testing Instructions:
1. Introduce network throttling.
2. Spin up NooBaa.
3. Run `aws s3api --endpoint <val> put-object ...` should run just fine provided the timeouts are correctly set.
